scripts/prepare_network.py

```diff
@@ -204,7 +205,9 @@ def enforce_autarky(n, only_crossborder=False):
                     ].index
     else:
         lines_rm = n.lines.index
-        links_rm = n.links.index
+        for i in n.links.index:
+            if n.links.loc[i,'carrier'] == 'DC':
+                links_rm.append(i)
```

Comment:
This is a more compact formulation:
```suggestion
        links_rm = n.links.loc[n.links.carrier=="DC"].index
```
You would also remove initialisation of `links_rm` further up.